pkg/domain/ipam.go

```diff
@@ -0,0 +1,31 @@
+package domain
+
+// Device represents a physical device with a network interface on the stored IP address.
+type Device struct {
+	ID     string
+	IP     string
+	Subnet string
+}
+
+// Subnet represents a block of IP addresses allocated to a ResourceOwner.
+type Subnet struct {
+	ID       string
+	Network  string
+	MaskBits int8
+	Location string
+	Customer string
```

Comment:
Question: should this be `CustomerID`? From a storage perspective we'll want this to be a foreign key to the customer primary key?

Reply:
Yes, and likewise below. I always waffle on naming in this situation.